common/src/main/java/com/wynntils/core/features/overlays/Overlay.java

```diff
@@ -23,7 +23,7 @@
 import net.minecraft.client.resources.language.I18n;
 import net.minecraft.world.phys.Vec2;
 
-public abstract class Overlay implements Translatable, Configurable {
+public abstract class Overlay implements Translatable, Configurable, Comparable<Overlay> {
```

Comment:
I agree with Qodana. We should implement `equals`.

Reply:
I don't know how we can improve on the default one? Do we want equals to not care about instances, but rather overlay types? I am not sure if that is better.

Reply:
Default equals just implements identity. We should compare the very same fields as we look at in the `compareTo` method. We should also implement a hash function that returns a hash based on those. That is "best practice" java, to not get caught in pitfalls. You don't have to do it now, but we should make sure we never implements `equals` or `hashCode` without the other, and never `compareTo` without the other two.